cleanlab/object_detection/summary.py

```diff
@@ -142,13 +142,9 @@ def visualize(
 
     if save_path:
         allowed_image_formats = set(["png", "pdf", "ps", "eps", "svg"])
-        image_format = (
-            save_path[-3:]
-            if len(save_path) > 2
-            and save_path[-4] == "."
-            and save_path[-3:] in allowed_image_formats
-            else None
-        )
+        image_format: Optional[str] = save_path.split(".")[-1]
+        if "." not in save_path or image_format not in allowed_image_formats:
+            image_format = None
```

Comment:
Would perhaps be nice to add a test to make sure that the following `plt.savefig()` call still works as expected when this `image_format=None`